cloudrun-malware-scanner/zip.js

```diff
@@ -0,0 +1,55 @@
+const yauzl = require('yauzl');
+const {logger} = require('./logger.js');
+
+// 10 min timeout for scanning.
+const CLAMD_TIMEOUT = 600000;
+
+/**
+ * @param {*} scanner
+ * @param {*} reader
+ * @return {Promise<Array<string>>} collection of clamd results
+ */
+function scanGcsZipFile(scanner, reader) {
+  const results = [];
+
+  return new Promise((resolve, reject) => {
+    yauzl.fromRandomAccessReader(
+        reader,
+        gcsFile.size,
+        {lazyEntries: true},
+        (err, zipfile) => {
+          if (err) {
+            throw err;
+          }
+
+          zipfile.readEntry();
+          zipfile.on('entry', (entry) => {
+            zipfile.openReadStream(entry, async (err, readStream) => {
+              if (err) throw err;
+
+              readStream.on('end', () => {
+                zipfile.readEntry();
+              });
+
+              readStream.on('error', (err) => {
+                throw err;
+              });
+
+              const result = await scanner.scanStream(
+                  readStream, CLAMD_TIMEOUT);
```

Comment:
nit
is this CLAMD_TIMEOUT required?
what if this is not supplied?

if so, can you pass in timeout thru scanGcsZipFile?

scanGcsZipFile(scanner, reader, timeoutPerFile)

to force the caller to decide (instead of hiding it)

Reply:
it would be great if we added some tests to validate these

Reply:
ah that's a good point. I'll pass it in. I think we just want some reasonable timeout and it is required but passed thru to clamd